src/networkremote/outgoingdatacreator.cpp

```diff
@@ -744,3 +745,59 @@ void OutgoingDataCreator::SearchFinished(int id) {
 
   qLog(Debug) << "SearchFinished" << req.id_ << req.query_;
 }
+
+void OutgoingDataCreator::SetMusicEextensions(const QString& music_extensions) {
+  music_extensions_.clear();
+  for (const QString& extension : music_extensions.split(",")) {
+    QString ext = extension.trimmed();
+    if (ext.size() > 0 && ext.size() < 8)  // no empty string, less than 8 char
+      music_extensions_ << ext;
+  }
+}
+
+void OutgoingDataCreator::SendListFiles(QString relative_path) {
+  pb::remote::Message msg;
+  msg.set_type(pb::remote::LIST_FILES);
+
+  pb::remote::ResponseListFiles* files = msg.mutable_response_list_files();
+
+  if (files_root_folder_.isEmpty())
+    files->set_error(pb::remote::ResponseListFiles::ROOT_DIR_NOT_SET);
```

Comment:
You should return early for these error states to make the following code more readable and less indented.

Reply:
alright. I know some people that hate returning before the end but I kind of prefer it too

Reply:
done